bokeh/client/session.py

```diff
@@ -153,11 +153,16 @@ def __init__(self, session_id=DEFAULT_SESSION_ID, io_loop=None, url=DEFAULT_SERV
         self._connection = ClientConnection(session=self, io_loop=io_loop, url=url)
 
         self._current_patch = None
+        self._callbacks = []
 
     def _attach_document(self, document):
         self._document = document
         self._document.on_change(self._document_changed)
 
+    def add_periodic_callback(self, cb, callback_time):
```

Comment:
I was thinking this method should be on Document - remember, the application can only modify documents. Document would keep a set of callbacks, and emit events for PeriodicCallbackAdded / PeriodicCallbackRemoved. ClientSession and ServerSession then need to watch the Document and add/remove callbacks when they are added/removed. The set of callbacks kept by Document should perhaps be things like:
```
 class PeriodicCallback(object):
   @property
   def id(self)
   @property
   def period(self)
   @property
   def callable(self)
```

Then that's the thing that goes in the `PeriodicCallbackAdded` and `PeriodicCallbackRemoved` events.